### ISSUE TITLE
Snippet editor: fields shouldn't look active when not focused

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#503]
In the new React snippet editor:
- click on a field
- then click outside of the field or on any other focusable element that is not a snippet editor field
- the field still looks "focused" while actually it isn't:

<img width="711" alt="screen shot 2018-05-07 at 10 47 49" src="https://user-images.githubusercontent.com/1682452/39693434-4a12151e-51e4-11e8-9529-aa54fea44f31.png">

**Comment by afercia** [comment]
It's very evident when tabbing from the description field to the "Close" button: the field still has a "focus style" and the button gets its focus style:

<img width="701" alt="screen shot 2018-05-07 at 11 13 54" src="https://user-images.githubusercontent.com/1682452/39694513-e1c1a160-51e7-11e8-8494-a5e5bc9a43cf.png">

**Comment by afercia** [comment]
Just noticed that sometimes, clicking on a preview "part" doesn't work well and there's the need to click twice to see the related field focused. E.g. after a first click, clicking on another preview part highlighted below doesn't do anything:

![screen shot 2018-05-07 at 11 54 57](https://user-images.githubusercontent.com/1682452/39696361-ceb1923c-51ed-11e8-8ce2-7d3ee762f3f5.jpg)

**Comment by afercia** [comment]
When touching this part of the code, please see also #532

**Comment by afercia** [comment]
Looking a bit more into this, there's also an usability issue:
- click on one of the field "labels"
- focus is moved into the field, and the caret is within the field

<img width="399" alt="screen shot 2018-05-18 at 11 37 59" src="https://user-images.githubusercontent.com/1682452/40227977-4682dbb8-5a90-11e8-8fa5-ecb8792d663f.png">

- click again the same label (in this case: "SEO title")
- the field will still look like it's focused but actually it isn't
- the caret disappears, it's not within the field any longer

<img width="380" alt="screen shot 2018-05-18 at 11 38 06" src="https://user-images.githubusercontent.com/1682452/40228036-74e9f11c-5a90-11e8-8d13-a5c670993a9c.png">

As an overall consideration, I understand the original intent of this implementation. However, trying to replicate programmatically native behaviors like hover, focus (and blur), is really hard and prone to subtle issues.
